edx_when/api.py

```diff
@@ -318,6 +377,14 @@ def set_date_for_block(course_id, block_id, field, date_or_timedelta, user=None,
     else:
         date_kwargs = {'abs_date': date_or_timedelta}
 
+    def _set_content_date_policy(date_kwargs, existing_content_date):
+        # Race conditions were creating multiple DatePolicies w/ the same values. Handle that case.
+        existing_policies = list(models.DatePolicy.objects.filter(**date_kwargs).order_by('id'))
```

Comment:
Just me wondering, and no obligation at all to test or try or change anything, but I wonder why do the list and [0] vs just using `.first()` on the query set. Not sure the error it throws if none (maybe none) but maybe an option. Not at all for you to do, just commenting while in my head.

Reply:
According to https://docs.djangoproject.com/en/2.2/ref/models/querysets/#first, using `first()` would work just as well. I simply copied the existing code, removing the never-thrown exception.